### PR TITLE
add new arguments to allow vpc/vpce endpoints as intrinsics

### PR DESCRIPTION
**Pull request by wchengru**
*Issue #, if available:*

*Description of changes:*

*Description of how you validated changes:*

*Checklist:*

- [x] Write/update tests
- [x] `make pr` passes
- [x] Update documentation
- [x] Verify transformed template deploys and application functions as expected

*Examples?*

Please reach out in the comments, if you want to add an example. Examples will be 
added to `sam init` through https://github.com/awslabs/aws-sam-cli-app-templates/

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: tests/swagger/test_swagger.py [hunk 1]
```diff
@@ -1200,7 +1208,90 @@ def test_must_add_vpc_deny(self):
...
+                },
+            ],
+        }
+
+        self.assertEqual(deep_sort_lists(expected), deep_sort_lists(self.editor.swagger[_X_POLICY]))
+
+
+    def test_must_add_vpc_allow_string_and_instrinic(self):
+
+        resourcePolicy = {
```
Comment (awood45) [inline tests/swagger/test_swagger.py:1224]: A couple changes:
- Please make sure the explicit intrinsic entries do NOT have "VPC" or "VPCE" in their strings, to ensure we don't regress on requiring regex matching.
- Please include more than one to ensure it's handling the full list.

Can we also test both whitelist and blocklist in a single test, to ensure both are applied?

File: samtranslator/swagger/swagger.py [hunk 2]
```diff
@@ -980,33 +1002,42 @@ def _add_ip_resource_policy_for_method(self, ip_list, conditional, resource_list
...
+                    vpc_list.append(endpoint)
+            if vpc_list:
+                condition.setdefault("aws:SourceVpc", []).extend(vpc_list)
+            if vpce_list:
+                condition.setdefault("aws:SourceVpce", []).extend(vpce_list)
+        if intrinsic_vpc_endpoint_list is not None:
+            condition.setdefault("aws:SourceVpc", []).extend(intrinsic_vpc_endpoint_list)
+        if intrinsic_vpce_endpoint_list is not None:
+            condition.setdefault("aws:SourceVpce", []).extend(intrinsic_vpce_endpoint_list)
+        if not condition:
```
Comment (awood45) [inline samtranslator/swagger/swagger.py:1038]: What is this new clause doing?

**Review by awood45 — CHANGES_REQUESTED** [review]
Test request, as well as a question.

File: samtranslator/swagger/swagger.py (hunk 2 above)
Comment (wchengru) [reply samtranslator/swagger/swagger.py:1038]: The plan here is adding a sanity check that it doesn't write empty endpoint lists into this section. It is similar as the original code does at [this line](https://github.com/aws/serverless-application-model/blob/develop/samtranslator/swagger/swagger.py#L989). 
I just realized that although the three lists in the `endpoint_dict` shouldn't be empty, but the regex classified vpc/vpce list may still be empty. So I modified this condition to better verify that the endpoint allow/deny lists are not empty.

File: tests/swagger/test_swagger.py (hunk 1 above)
Comment (wchengru) [reply tests/swagger/test_swagger.py:1224]: Updated intrinsic entries and added more test cases.

**Review by awood45 — APPROVED** [review]

**Pull request merged**